### ISSUE TITLE
Question : is this possible to configure default value column from comments ?

### ISSUE THREAD

**Issue by eddycharly** [norwoodj/helm-docs#22]
Can we configure what is set in the default value column from comments ?
Sometimes the default value is quite large (a configuration file for example) and i would like to put a short description like `see in values.yaml` rather than the raw whole config file content.

**Comment by Grannath** [comment]
Have the same feature request. Sometimes I calculate the default from other values, so I cannot put it into the values.yaml directly.

**Comment by norwoodj** [comment]
Hmmm, apologies @eddycharly for taking so long to reply. I wasn't entirely sure what the use case was at first, and I kinda forgot to come back to this. This makes sense to me. Can y'all comment here with a quick example of the comment format you're envisioning to accomplish this? Definitely think this could be easily accomplished by having helm-docs recognize a new comment format and perform more complicated logic like you're describing.

**Comment by eddycharly** [comment]
@norwoodj see an example below.

Let's say i have an entry in my values.yaml like this :
```yaml
pipeline:
  config:
    # pipeline.config.logging -- Pipeline configuration for logging
    logging:
      loglevel.controller: info
      loglevel.webhook: info
      zap-logger-config: |
        {
          "level": "info",
          "development": false,
          "sampling": {
            "initial": 100,
            "thereafter": 100
          },
          "outputPaths": ["stdout"],
          "errorOutputPaths": ["stderr"],
          "encoding": "json",
          "encoderConfig": {
            "timeKey": "",
            "levelKey": "level",
            "nameKey": "logger",
            "callerKey": "caller",
            "messageKey": "msg",
            "stacktraceKey": "stacktrace",
            "lineEnding": "",
            "levelEncoder": "",
            "timeEncoder": "",
            "durationEncoder": "",
            "callerEncoder": ""
          }
        }
```

this will generate the following markdown (more or less, i use a custom template) :

| Key | Type | Description | Default |
|-----|------|-------------|---------|
| pipeline.config.logging | object | Pipeline configuration for logging | `{"loglevel.controller":"info","loglevel.webhook":"info","zap-logger-config":"{\n  \"level\": \"info\",\n  \"development\": false,\n  \"sampling\": {\n    \"initial\": 100,\n    \"thereafter\": 100\n  },\n  \"outputPaths\": [\"stdout\"],\n  \"errorOutputPaths\": [\"stderr\"],\n  \"encoding\": \"json\",\n  \"encoderConfig\": {\n    \"timeKey\": \"\",\n    \"levelKey\": \"level\",\n    \"nameKey\": \"logger\",\n    \"callerKey\": \"caller\",\n    \"messageKey\": \"msg\",\n    \"stacktraceKey\": \"stacktrace\",\n    \"lineEnding\": \"\",\n    \"levelEncoder\": \"\",\n    \"timeEncoder\": \"\",\n    \"durationEncoder\": \"\",\n    \"callerEncoder\": \"\"\n  }\n}\n"}` |

The default value is the raw object making the docs hard to read. For such a complex object i would rather just put a link to the values.yaml file, something like :

| Key | Type | Description | Default |
|-----|------|-------------|---------|
| pipeline.config.logging | object | Pipeline configuration for logging | see [values.yaml](./values.yaml) |

About the comment format, maybe it could be something like this ?

```yaml
pipeline:
  config:
    # pipeline.config.logging -- Pipeline configuration for logging -- see [values.yaml](./values.yaml)
    logging:
      loglevel.controller: info
      loglevel.webhook: info
      zap-logger-config: |
        {
          "level": "info",
          "development": false,
          "sampling": {
            "initial": 100,
            "thereafter": 100
          },
          "outputPaths": ["stdout"],
          "errorOutputPaths": ["stderr"],
          "encoding": "json",
          "encoderConfig": {
            "timeKey": "",
            "levelKey": "level",
            "nameKey": "logger",
            "callerKey": "caller",
            "messageKey": "msg",
            "stacktraceKey": "stacktrace",
            "lineEnding": "",
            "levelEncoder": "",
            "timeEncoder": "",
            "durationEncoder": "",
            "callerEncoder": ""
          }
        }
```

**Comment by Grannath** [comment]
I would prefer to split it. Lines are already getting quite long for some values.

```yaml
pipeline:
  config:
    # pipeline.config.logging -- Pipeline configuration for logging
    # default -- see [values.yaml](./values.yaml)
    logging:
      [...]
```

This could be extended, to also allow customizing other columns, or even add new columns. So, when parsing, check all comments. Treat anything before the two dashes as a column name (ignore case). Special case is the field name, which maps to description. Maybe require that one to be present for other comments to be applied as well. Also maybe include a flag to disable custom columns by default and instead give an error for unknown columns, to prevent user errors on typos.

**Comment by eddycharly** [comment]
@Grannath 👍

**Comment by Grannath** [comment]
@eddycharly I messed around with this in a merge request. Could need some review, since I have pretty much zero experience with Go. Also needs more examples, I guess. Please have a look and leave comments!

**Comment by norwoodj** [comment]
Hey, @eddycharly and @Grannath I'm really sorry I've been so long to respond to this. I recently started a new job and have had a lot going on in my personal life

**Comment by norwoodj** [comment]
Also, hit the wrong button. I didn't mean to close this issue.

**Comment by eddycharly** [comment]
Hey @norwoodj, I had time to work on this today, I should be able to open a PR for it tomorrow. Just letting you know, in case it helps getting your attention 😄

**Comment by norwoodj** [comment]
@eddycharly, I'm really sorry, I'm the worst open source owner ever. I am finally looking at this now!

**Comment by eddycharly** [comment]
No worries, only 5 days this time ! ;-) 👍

**Comment by norwoodj** [comment]
I hope to be better going forward. I merged your change, it was fantastic @eddycharly. I just released [v0.10.0](https://github.com/norwoodj/helm-docs/releases/tag/v0.10.0) with your update. Thanks again!
